addons/website_sale/models/product.py

```diff
@@ -156,9 +156,8 @@ def _website_price(self):
         partner = self.env.user.partner_id
         pricelist = request.website.get_current_pricelist()
 
-        self2 = self.with_context(pricelist=pricelist.id, partner=partner)
-        if self2._context == self._context:
-            self2 = self
+        context = dict(self._context, pricelist=pricelist.id, partner=partner)
+        self2 = self._context != context and self.with_context(context) or self
```

Comment:
I prefer the ternary operator:
self2 = self.with_context(context) if self._context != context else self